backend/src/api.py

```diff
@@ -433,5 +457,53 @@ def add_package(
             name=name,
             description=description,
             dependencies=dependencies or [],
+            icon=icon,
+            color=color,
         )
     )
+
+
+@dataclass
+class ToggleSetting:
+    label: str
+    key: str
+    description: str
+    default: bool = False
+    disabled: bool = False
+    type: str = "toggle"
+
+
+DropdownOption = TypedDict("DropdownOption", {"label": str, "value": str})
```

Comment:
You're supposed to use `TypedDict` like this:

```suggestion
class DropdownOption(TypedDict):
    label: str
    value: str
```